Update dependencies, modernize

mook-as opened the pull request:
- Convert to `dep` instead of `godep`
- Update dependencies (because the conversion :arrow_up: isn't totally clean)
- Build for linux/windows i386 as well
- Use `code.cloudfoundry.org` over `github.com/cloudfoundry`
- Use `printf` for the `make/` scripts (because not all shells support escaping without `echo -e`)

Review locally via:
- `git diff origin/master -- . ':!/vendor/'`
- `git diff --stat origin/master`

andreas-kupries commented on make/dist:
```diff
@@ -7,10 +7,11 @@ set -o errexit
 . make/include/colors.sh
 . make/include/versioning.sh
 
-echo "${OK_COLOR}==> Disting all${NO_COLOR}"; \
```
Can we use `Distributing` instead of `Disting` ?

andreas-kupries commented on make/format:
```diff
@@ -4,20 +4,15 @@ set -o errexit
 
 . make/include/colors.sh
 
-echo "${OK_COLOR}==> Formating ${ERROR_COLOR}"
```
`Formatting` (double-t)

mook-as commented:
Fixed in 6beaee650c8d13eb224021438810e3fa289ae856

I changed the signature of `CCResources.GetResources` to return a slice instead of a pointer to a slice; the `len()` of a nil slice is `0`, and running `range` over it is fine (with 0 iterations).
